pyHS100/pyHS100.py

```diff
@@ -283,6 +297,168 @@ def identify(self):
+        Returns device icon
+        Note: this doesn't seem to work when not using the cloud service, not tested with it either.
+        :return: icon and its hash
+        :rtype: dict
+        :raises SmartPlugException: on error
+        """
+        return self._query_helper("system", "get_dev_icon")
+
+    @icon.setter
+    def icon(self, icon):
+        """
+        Content for hash and icon are unknown.
+        :param str icon: Icon path(?)
+        :raises NotImplementedError: when not implemented
+        :raises SmartPlugError: on error
+        """
+        raise NotImplementedError("Values for this call are unknown at this point.")
+        # here just for the sake of completeness
+        # self._query_helper("system", "set_dev_icon", {"icon": "", "hash": ""})
+
+    @property
+    def time(self):
+        """
+        Returns current time from the device.
+        :return: datetime for device's time
+        :rtype: datetime.datetime
+        :raises SmartPlugException: on error
+        """
+        response = self._query_helper("time", "get_time")
+        return datetime.datetime(response["year"], response["month"], response["mday"], response["hour"], response["min"], response["sec"])
+
+    @time.setter
+    def time(self, ts):
+        """
+        Sets time based on datetime object.
+        Note, this calls set_timezone
+        :param datetime.datetime ts: New date and time
+        :return: result
+        :type: dict
+        :raises NotImplemented: when not implemented.
+        :raises SmartPlugException: on error
+        """
+        raise NotImplementedError("Setting time does not seem to work on HS110 although it returns no error.")
+        """ here just for the sake of completeness / if someone figures out why it doesn't work.
+        ts_obj = {
+            "index": self.timezone["index"],
+            "hour": ts.hour,
+            "min": ts.minute,
+            "sec": ts.second,
+            "year": ts.year,
+            "month": ts.month,
+            "mday": ts.day,
+        }
+
+        return self._query_helper("time", "set_timezone", ts_obj)
+        """
+
+    @property
+    def timezone(self):
+        """
+        Returns timezone information
+        :return: Timezone information
+        :rtype: dict
+        :raises SmartPlugException: on error
+        """
+        return self._query_helper("time", "get_timezone")
+
+    @property
+    def hw_info(self):
+        """
+        Returns information about hardware
+        :return: Information about hardware
+        :rtype: dict
+        """
+        keys = ["sw_ver", "hw_ver", "mac", "hwId", "fwId", "oemId", "dev_name"]
+        return {key: self.sys_info[key] for key in keys}
+
+    @property
+    def on_since(self):
+        """
+        Returns pretty-printed on-time
+        :return: datetime for on since
+        :rtype: datetime
+        """
+        return datetime.datetime.now() - datetime.timedelta(seconds=self.sys_info["on_time"])
```

Comment:
We should adhere to PEP8 to guarantee consistently formatted code, and the line limit ensures even lower resolution devices can display the code in a reasonable manner.